spec/controllers/v1/actions_controller_spec.rb

```diff
@@ -2,20 +2,32 @@
 
 RSpec.describe V1::ActionsController, type: :controller do
   describe "#create" do
-    before do
-      FactoryGirl.create(:activity, template_id: 'real_id')
-      FactoryGirl.create(:person, phone: '510-555-4444', email: 'joe@example.com')
+    render_views
+    context "with good params" do
+      before do
+        FactoryGirl.create(:activity, template_id: 'real_id')
+        FactoryGirl.create(:person, uuid: 'the-uuid', email: 'joe@example.com')
+      end
+      it "returns person object" do
+        expect(Person).to receive(:create_or_update).with(email: 'joe@example.com')
+          .and_call_original
+        post :create, person: { email: 'joe@example.com' }, template_id: 'real_id'
+        parsed_response = JSON.parse(response.body)
+        expect(parsed_response.slice('uuid', 'completed_activities').values)
+          .to eq ["the-uuid", ["real_id"]]
+      end
     end
-    it "creates action with good parameters" do
-      expect(Person).to receive(:create_or_update).with(uuid: 'the_uuid', phone: '510-555-4444')
-        .and_call_original
-      post :create, person: { uuid: 'the_uuid', phone: '510-555-4444' }, template_id: 'real_id'
-      expect(JSON.parse(response.body)['success']).to be true
-    end
-    it "doesn't create action if missing parameters" do
-      post :create, template_id: 'real_id'
-      expect(JSON.parse(response.body)['success']).to be_nil
-      expect(JSON.parse(response.body)['error']).to eq 'person is required'
+    context "with bad params" do
+      it "returns error if person is missing" do
+        post :create, template_id: 'real_id'
+        expect(JSON.parse(response.body)['success']).to be_nil
+        expect(JSON.parse(response.body)['error']).to eq 'person is required'
+      end
+      it "returns error if person is incomplete or template_id is bad" do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.